cmd/zed/agents/zfs_retire.c

```diff
@@ -523,6 +523,9 @@ zfs_retire_recv(fmd_hdl_t *hdl, fmd_event_t *ep, nvlist_t *nvl,
 		} else if (fmd_nvl_class_match(hdl, fault,
 		    "fault.fs.zfs.vdev.checksum")) {
 			degrade_device = B_TRUE;
+		} else if (fmd_nvl_class_match(hdl, fault,
+		    "fault.fs.zfs.vdev.slow_io")) {
+			degrade_device = B_TRUE;
```

Comment:
So is `slow_io` a new event type, or am I misreading this?  If it is, how does it differ from the current `delay` events?:
```
#define FM_EREPORT_ZFS_DELAY            "delay"
```

Reply:
`fault.fs.zfs.vdev.slow_io` is a new **fault** diagnosis (sent by diagnosis and consumed by the retire agent).
and
`FM_EREPORT_ZFS_DELAY` is an **event** that is sent from the kernel which is consumed by the diagnosis engine